File: _25_1000DigitFibonacciNumber.java
import java.util.*;
import java.math.BigInteger;

public class _25_1000DigitFibonacciNumber{
    
    public static void main(String[] args){
        int index = 2; // F_1 = 1, F_2 = 1, F_3 = 2, ...
        BigInteger prevFib = BigInteger.ONE;
        BigInteger currFib = BigInteger.ONE;
        while (currFib.toString().length() < 1000){
            BigInteger temp = currFib;
            currFib = currFib.add(prevFib);
            prevFib = temp;
            index++;
        }
        System.out.println("The index of the first Fibonacci term to contain at least 1000 digits is " + index); // 4782
    }
}

File: c#/_25_1000DigitFibonacciNumber.cs
using System;
using System.Numerics;

public class _25_1000DigitFibonacciNumber{
	public static int Euler25(){
		int idx = 1;
		BigInteger f1 = 0;
		BigInteger f2 = 1;
		while (f2.ToString().Length < 1000){
			BigInteger temp = f2;
			f2 += f1;
			f1 = temp;
			idx++;
		}
		return idx;
	}
	
	public static void Main(){
		Console.WriteLine(Euler25());
	}
}
